driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -196,17 +193,13 @@ private PoolingOptions options() {
         return manager.configuration().getPoolingOptions();
     }
 
-    public Connection borrowConnection(long timeout, TimeUnit unit) throws ConnectionException, TimeoutException {
+    ListenableFuture<Connection> borrowConnection(int maxQueueSize) {
         Phase phase = this.phase.get();
         if (phase != Phase.READY)
-            // Note: throwing a ConnectionException is probably fine in practice as it will trigger the creation of a new host.
-            // That being said, maybe having a specific exception could be cleaner.
-            throw new ConnectionException(host.getSocketAddress(), "Pool is " + phase);
+            return Futures.immediateFailedFuture(new ConnectionException(host.getSocketAddress(), "Pool is " + phase));
 
         if (connections.isEmpty()) {
-            if (!host.convictionPolicy.canReconnectNow())
-                throw new TimeoutException("Connection pool is empty, currently trying to reestablish connections");
-            else {
+            if (host.convictionPolicy.canReconnectNow()) {
                 int coreSize = options().getCoreConnectionsPerHost(hostDistance);
                 if (coreSize == 0) {
                     maybeSpawnNewConnection();
```

Comment:
Not related to that ticket but I don't get that line, if users set the "core connections pool size" to 0, is the driver creating a new connection for all queries until the max number is reached?